govcd/vdc.go

```diff
@@ -920,7 +920,7 @@ func (vdc *Vdc) QueryVappVmTemplate(catalogName, vappTemplateName, vmNameInTempl
 	// this allows to query deployed and not deployed templates
 	results, err := vdc.QueryWithNotEncodedParams(nil, map[string]string{"type": queryType,
 		"filter": "catalogName==" + url.QueryEscape(catalogName) + ";containerName==" + url.QueryEscape(vappTemplateName) + ";name==" + url.QueryEscape(vmNameInTemplate) +
-			";isVAppTemplate==true;status!=FAILED_CREATION;status!=UNKNOWN;status!=UNRECOGNIZED;status!=UNRESOLVED&links=true;",
+			";isVAppTemplate==true;status!=FAILED_CREATION;status!=UNKNOWN;status!=UNRECOGNIZED;status!=UNRESOLVED;status!=LOCAL_COPY_UNAVAILABLE&links=true;",
```

Comment:
I think also this change should be reconsidered, as it may have side effects: we sometimes search vApp templates for the purpose of synchronizing them. If this function only returns the synchronized ones, we will never get the job done

Reply:
Agree. I have done a separated function instead. The original `QueryVappVmTemplate` remains the same (with some extra docs), and I added `QueryVappSynchronizedVmTemplate` which invokes the former and checks that the `State` is the correct one.